pkg/segment/writer/agiletree.go

```diff
@@ -246,13 +250,14 @@ func (stb *StarTreeBuilder) DropColumns(colsToDrop []string) error {
 	stb.segDictEncRev = toputils.RemoveElements(stb.segDictEncRev, dropIndexes)
 	stb.segDictLastNum = toputils.RemoveElements(stb.segDictLastNum, dropIndexes)
 
-	// No need to update wipRecNumToColEnc, since it will be reset in each ComputeStarTree based on groupByKeys
+	// No need to update wipRecNumToColEnc based on index, since it will be repopulated based on update groupByKeys in creatEnc
 	stb.wipRecNumToColEnc = stb.wipRecNumToColEnc[:stb.numGroupByCols]
```

Comment:
I'm still confused
1. What does "based on index" mean? I don't see a variable `index`
2. You're not calling `creatEnc()` in this (public) function though, so you don't know that the caller will invoke `creatEnc()`. If the caller doesn't, will this tree be in an invalid state?

Reply:
Above members are cleaned up based on the index of dropped columns. This particular member doesn't need to be cleaned up since it is repopulated in ComputeStarTree. Have added comment to clarify this.